keyboards/aki27/cocot46plus/keymaps/default/keymap.c

```diff
@@ -0,0 +1,134 @@
+                                                                 XXXXXXX, KC_MS_BTN3,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX
+                                                            //`--------------'  `--------------'
+    ),
+  [_TRACKBALL] = LAYOUT(
+  //|-------------------------------------------------------|                                   |-------------------------------------------------------|
+      QK_BOOT, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, RGB_TOG,                                       SCRL_TO,  CPI_SW, SCRL_SW, ROT_L15, ROT_R15, XXXXXXX,
+  //|-------------------------------------------------------|                                   |-------------------------------------------------------|
+      XXXXXXX, XXXXXXX, RGB_VAI, RGB_SAI, RGB_HUI, RGB_MOD,                                       SCRL_MO, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+  //|-------------------------------------------------------|                                   |-------------------------------------------------------|
+      XXXXXXX, XXXXXXX, RGB_VAD, RGB_SAD, RGB_HUD,RGB_RMOD,                                       SCRL_IN, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+  //|-------------------------------------------------------|                                   |-------------------------------------------------------|
+                        KC_LGUI, DEL_ALT, KC_TRNS,  KC_SPC,   KC_MS_BTN1,             KC_MS_BTN2,  KC_ENT, RS_HENK, KC_BSPC,  KC_ESC,
+                                                                 XXXXXXX, KC_MS_BTN3,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX
+                                                            //`--------------'  `--------------'
+    )
+};
+
+// Same function on all layers for now.
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN) },
+    [1] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN) },
+    [2] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN) },
+    [3] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN) },
+};
+#endif
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    switch (get_highest_layer(state)) {
+    case _LOWER:
+        #ifdef RGBLIGHT_ENABLE
+        rgblight_sethsv_range(HSV_BLUE, 0, 2);
+        #endif
+        cocot_set_scroll_mode(true);
+        break;
+    case _RAISE:
+        #ifdef RGBLIGHT_ENABLE
+        rgblight_sethsv_range(HSV_RED, 0, 2);
+        #endif
+        cocot_set_scroll_mode(true);
+        break;
+    case _TRACKBALL:
+        #ifdef RGBLIGHT_ENABLE
+        rgblight_sethsv_range(HSV_GREEN, 0, 2);
+        #endif
+        cocot_set_scroll_mode(false);
+        break;
+    default:
+        #ifdef RGBLIGHT_ENABLE
+        rgblight_sethsv_range( 0, 0, 0, 0, 2);
+        #endif
+        cocot_set_scroll_mode(false);
+        break;
+    }
+    #ifdef RGBLIGHT_ENABLE
+    rgblight_set_effect_range( 2, 10);
+    #endif
+      return state;
+};
+
+
+#ifdef OLED_ENABLE
+bool oled_task_user(void) {
```

Comment:
This needs to be moved to the kb level, and as:
```c
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }
    // ... rest of code
}
```

Reply:
Thanks for the feedback, @drashna 

I've pushed a commit that does this now.